examples/gh-runner-gke-dind/README.md

```diff
@@ -0,0 +1,76 @@
+## Self Hosted runners on GKE that support Docker workflows
+
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|:----:|:-----:|:-----:|
+| gh\_token | Github token that is used for generating Self Hosted Runner Token | string | n/a | yes |
+| project\_id | The project id to deploy Github Runner MIG | string | n/a | yes |
+| repo\_name | Name of the repo for the Github Action | string | n/a | yes |
+| repo\_owner | Owner of the repo for the Github Action | string | n/a | yes |
+| repo\_url | Repo URL for the Github Action | string | n/a | yes |
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| ca\_certificate | The cluster ca certificate (base64 encoded) |
+| client\_token | The bearer token for auth |
+| cluster\_name | Cluster name |
+| kubernetes\_endpoint | The cluster endpoint |
+| location | Cluster location |
+| service\_account | The default service account used for running nodes. |
+
+ <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+### Steps to deploy this example
+
+- Step 1: Create terraform.tfvars file with the necessary values
+
+Access tokens require repo scope for private repos and public_repo scope for public repos. GitHub Apps must have the administration permission to use this API. Authenticated users must have admin access to the repository to use this API.
+
+More info can be found [here](https://developer.github.com/v3/actions/self_hosted_runners/)
+
+```
+project_id = "your-project-id"
+gh_token   = "your-github-token"
+repo_url   = "https://github.com/owner/your-repo-name"
+repo_name  = "your-repo-name"
+repo_owner = "owner"
+```
+
+- Step 2: Create the infrastructure
+
+```
+terraform init
+terraform plan
+terraform apply
+```
+
+- Step 3: Build and push the docker image
+
+```sh
+docker build --tag gcr.io/${PROJECT_ID}/runner:0.1 --tag gcr.io/${PROJECT_ID}/runner:latest .
```

Comment:
An alternative to consider using here is `gcloud builds submit`. That would alleviate the need for the user to have to setup a local docker daemon.

Reply:
+1. I'd rather take a dependency on Cloud Build here if you think that makes sense.

Reply:
Done, added cloudbuild.yamls for all examples and updated instructions.